bin/fiberassign

```diff
@@ -281,6 +299,34 @@ def add_fiber_data_columns(tile_data, fiberassign_data, mtl_data, sky_data, posi
     
     return np.array(fiberassign_data)
 
+
+def make_target_hdu_data(fiberassign_data, mtl_data):
+    new_target_data = fiberassign_data.copy()
+    n_objects = len(new_target_data)
+    
+    print('  adding all columns from mtl data in make_target_hdu_data')
+    print(' Total of MTL points {}. Total of new_target_data {}'.format(
+            len(mtl_data), len(new_target_data)))
+    columns=['TARGETID','SUBPRIORITY', 'BRICKID', 'BRICK_OBJID', 'REF_ID',
+             'FLUX_G', 'FLUX_R', 'FLUX_Z', 'FLUX_W1', 'FLUX_W2', 
+             'MW_TRANSMISSION_G', 'MW_TRANSMISSION_R', 'MW_TRANSMISSION_Z',
+             'PSFDEPTH_G', 'PSFDEPTH_R', 'PSFDEPTH_Z',
+             'GALDEPTH_G', 'GALDEPTH_R', 'GALDEPTH_Z', 
+             'SHAPEDEV_R', 'SHAPEDEV_E1', 'SHAPEDEV_E2', 
+             'SHAPEEXP_R', 'SHAPEEXP_E1', 'SHAPEEXP_E2',
+             'FIBERFLUX_G', 'FIBERFLUX_R', 'FIBERFLUX_Z', 
+             'FIBERTOTFLUX_G', 'FIBERTOTFLUX_R', 'FIBERTOTFLUX_Z', 'HPXPIXEL']
```

Comment:
My intension for the TARGETS HDU was that it would be a literal copy of the MTL rows, reordered to match the order of the fiber assignments, without any hardcoded list of expected columns.  That way if the MTL adds or removed columns in the future, they automatically get propagated without also requiring updates to fiberassign.  i.e. let the input MTL itself define which columns should be propagated (all of them), and then we'll update the datamodel documentation to match.

Is there some technical reason why you needed to have an explicit list of columns, or could you just propagate all of them?

Reply:
I did that to avoid duplicates in the columns (i.e. DESI_TARGET and DESI_TARGET_1).

Related question: for the 5000 fibers some targets will come from mtl but others will come from sky. Should I then only include the targets that come from mtl? Right now after the join the targets that come from sky are masked.